modules/core/src/main/scala/catcheffect/IOCatch.scala

```diff
@@ -0,0 +1,20 @@
+package catcheffect
+
+import Catch.catchForIO
+import cats.effect.*
+
+object IOCatch {
+  
+  def ioCatching[E]: IOCatchPartiallyApplied[E] =
```

Comment:
Alternatively can be named `catchingIO`. Thoughts?

Reply:
Do you have an opinion on `apply`?
```scala
IOCatch[Error] { h =>
  // ...
}
```